test/e2e/density.go

```diff
@@ -0,0 +1,178 @@
+	rc, err := c.ReplicationControllers(ns).Create(&api.ReplicationController{
+		ObjectMeta: api.ObjectMeta{
+			Name: name,
+		},
+		Spec: api.ReplicationControllerSpec{
+			Replicas: replicas,
+			Selector: map[string]string{
+				"name": name,
+			},
+			Template: &api.PodTemplateSpec{
+				ObjectMeta: api.ObjectMeta{
+					Labels: map[string]string{"name": name},
+				},
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:  name,
+							Image: image,
+							Ports: []api.Port{{ContainerPort: 80}},
+						},
+					},
+       				},
+			},
+		},
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	defer func () {
+		By("Cleaning up the replication controller")
+		rc.Spec.Replicas = 0
+		if _, err = c.ReplicationControllers(ns).Update(rc); err != nil {
+			Logf("Failed to resize replication controller %s to zero: %v", name, err)
+		}
+
+		// Delete the replication controller.
+		if err = c.ReplicationControllers(ns).Delete(name); err != nil {
+			Logf("Failed to delete replication controller %s: %v", name, err)
+		}
+	}()
+
+	By(fmt.Sprintf("Making sure all %d replicas exist", replicas))
+	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": name}))
+	pods, err := c.Pods(ns).List(label)
+	Expect(err).NotTo(HaveOccurred())
+	fail_cnt := 5
+	for same < fail_cnt {
+		current = len(pods.Items)
+		Logf("Controller %s: Found %d pods out of %d", name, current, replicas)
+		if len(pods.Items) == replicas {
+			break
+		} else if last < current {
+			same = 0
+		} else if last == current {
+			same++
+		} else if current < last {
+			Failf("Controller %s: Number of submitted pods dropped from %d to %d", last, current)
+		}
+
+		if same >= fail_cnt {
+			Logf("No pods submitted for the last %d checks", fail_cnt)
+		}
+		
+		last = current
+		time.Sleep(5 * time.Second)
+		pods, err = c.Pods(ns).List(label)
+		Expect(err).NotTo(HaveOccurred())
+	}
+	Expect(current).To(Equal(replicas))
+
+	By("Waiting for each pod to be running")
+	same = 0
+	last = 0
+	fail_cnt = 5
+	unknown := 0
+	pending := 0
+	for same < fail_cnt {
+		current = 0
+		pending = 0
+		unknown = 0
+		time.Sleep(5 * time.Second)
+		for _, pod := range pods.Items {
+			p, err := c.Pods(ns).Get(pod.Name)
+			Expect(err).NotTo(HaveOccurred())
+			if p.Status.Phase == api.PodRunning {
+				current++
+			} else if p.Status.Phase == api.PodPending {
+				pending++
+			} else if p.Status.Phase == api.PodUnknown {
+				unknown++
+			}
+		}
+		Logf("Pod States: %d running, %d pending, %d unknown ", current, pending, unknown)
+		if current == replicas {
+			break
+		} else if last < current {
+			same = 0
+		} else if last == current {
+			same++
+		} else if current < last {
+			Failf("Number of running pods dropped from %d to %d", last, current)
+		}
+		if same >= fail_cnt {
+			Logf("No pods started for the last %d checks", fail_cnt)
+		}
+		last = current
+	}
+	Expect(current).To(Equal(replicas))
+}
+
+var _ = PDescribe("Density", func() {
```

Comment:
Yeah, that's all I was asking. I didn't know if it was a manual process of removing the 'P', or if Ginkgo was smart enough that it wouldn't run it on a regexp, but if you focused on it explicitly, it would.